performance/tests/Button.test.js

```diff
@@ -1,82 +1,139 @@
-const {FPS, Mount, Update} = require('../TraceModel');
-const {getFileName} = require('../utils');
 const TestResults = require('../TestResults');
+const {CLS, DCL, FCP, FID, FPS, LCP} = require('../TraceModel');
+const {getFileName} = require('../utils');
 
 describe('Button', () => {
+	const component = 'Button';
+	TestResults.newFile(component);
+
 	describe('click', () => {
 		it('animates', async () => {
-			const filename = getFileName('Button');
+			const FPSValues = await FPS();
 			await page.goto('http://localhost:8080/button');
-			await page.tracing.start({path: filename, screenshots: false});
-			await page.waitFor(500);
-
+			await page.waitForSelector('#button');
 			await page.click('#button'); // to move mouse on the button.
 			await page.mouse.down();
-			await page.waitFor(200);
+			await page.waitForTimeout(100);
 			await page.mouse.up();
 			await page.mouse.down();
-			await page.waitFor(200);
+			await page.waitForTimeout(100);
 			await page.mouse.up();
 			await page.mouse.down();
-			await page.waitFor(200);
+			await page.waitForTimeout(100);
 			await page.mouse.up();
 			await page.mouse.down();
-			await page.waitFor(200);
+			await page.waitForTimeout(100);
 			await page.mouse.up();
 
-			await page.tracing.stop();
-
-			const actualFPS = FPS(filename);
-			TestResults.addResult({component: 'Button', type: 'Frames Per Second', actualValue: actualFPS});
-
-			const actualUpdateTime = Update(filename, 'Touchable');
-			TestResults.addResult({component: 'Button', type: 'Update', actualValue: actualUpdateTime});
+			const averageFPS = (FPSValues.reduce((a, b) => a + b, 0) / FPSValues.length) || 0;
```

Comment:
The formula to find the average FPS is always repeated. Shouldn't this be the return value of FPS()?

Reply:
Yes. I think we can decide the minimum fps. That's why I suggested 20 as a personal opinion.